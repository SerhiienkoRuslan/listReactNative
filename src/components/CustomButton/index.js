import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

import uiElementSize from 'constants/uiElementSize';
import colors from 'styles/colors';

const CustomButton = ({ onPress, text, customStyles, ...rest }) => {
  return (
    <TouchableOpacity
      {...rest}
      style={[styles.submitBtn, customStyles]}
      onPress={onPress}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  submitBtn: {
    height: uiElementSize.HEIGHT_INPUTS,
    backgroundColor: colors.primary,
    borderRadius: uiElementSize.BORDER_RADIUS,
    alignItems: 'center',
    justifyContent: 'center'
  },
  text: {
    color: 'white'
  }
});

CustomButton.defaultProps = {
  onPress: () => {},
  text: ''
};

export default CustomButton;
