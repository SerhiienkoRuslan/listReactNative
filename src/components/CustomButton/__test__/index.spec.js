import CustomButton from '../index';

describe('CustomButton component', () => {
  it('should render CustomButton component with props', () => {
    const component = shallow(<CustomButton text="Delete" />);
    expect(component).toMatchSnapshot();
  });

  it('should render CustomButton component without props', () => {
    const component = shallow(<CustomButton />);
    expect(component).toMatchSnapshot();
  });
});
