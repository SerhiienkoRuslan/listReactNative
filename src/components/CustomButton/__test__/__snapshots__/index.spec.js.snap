// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component should render CustomButton component with props 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fb5b5a",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    Delete
  </Text>
</ForwardRef>
`;

exports[`CustomButton component should render CustomButton component without props 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fb5b5a",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  />
</ForwardRef>
`;
